Yoast/wordpress-seo
# Reactify Advanced Settings tab

## Context
This PR removes the old PHP Advanced Settings tab and adds a new Advanced Settings tab that is written in React.

This makes it easier to include the Tab in the modal.

There is some discussion in https://github.com/Yoast/wordpress-seo/pull/14435 that might be interesting when reviewing this PR.


* This PR removes the PHP Advanced Settings tab and creates a new tab in React.

## Summary


This PR can be summarized in the following changelog entry:

* The Advanced Settings tab has been converted to React.
* Slightly new styling of the input fields.

## Test instructions

This PR can be tested by following these steps:
**The "new" advanced settings tab should behave exactly the same as the old advanced settings tab**
* Build everything and check the Advanced Settings tab.
	- [ ] Make sure it saves everything, and input fields look consistent and adhere to https://sketch.cloud/s/r0zwq/a/8ZmnPq.
	- [x] Make sure that the values written to and read from hidden inputs, this can be tested in the following way:
		1. Change an input.
		2. Save the post.
		3. Reload the page.
		4. Open the advanced tab, the value should still be present.
	- [x] Check all occurrences of the Advanced Tab
		- Posts
		- Pages
		- Tags
		- Categories
		- ???
	- [x] Make sure that the Private blog warning is displayed if the site is private. This can be set in `Settings`->`Reading`:
<img width="599" alt="afbeelding" src="https://user-images.githubusercontent.com/8782235/79347863-766b6c80-7f34-11ea-99e4-0d9007589ab0.png">

	

## For QA
Some IDs of the input fields are changed. I'll list the changes below. The old IDs are still present in the pages. However, they are only used for storing the value and not for any interaction with the user since they are hidden.

### For Post type pages
- `yoast_wpseo_meta-robots-noindex` -> `yoast_wpseo_meta-robots-noindex-react`
- `yoast_wpseo_meta-robots-nofollow` -> `yoast_wpseo_meta-robots-nofollow-react`
- `yoast_wpseo_meta-robots-adv` -> `yoast_wpseo_meta-robots-adv-react`
- `yoast_wpseo_bctitle` -> `yoast_wpseo_bctitle-react`
- `yoast_wpseo_canonical` -> `yoast_wpseo_canonical-react`

### For Taxonomy type pages
- `wpseo_noindex` -> `yoast_wpseo_meta-robots-noindex-react`
- `wpseo_bctitle` -> `yoast_wpseo_bctitle-react`
- `wpseo_canonical` -> `yoast_wpseo_canonical-react`

## Tasks
- [x] Ask Omar for clean-up
- [ ] Imports from packages should come from the root.
	- Export the new components from the root of `@yoast/components`
	- Export the helper functions from the root of `@yoast/helpers`


## UI changes

## Documentation

## Quality assurance


Fixes #14433
